Add option "lock-config" for shell command "config:set"

avstudnitz opened the pull request:
This is similar to the "lock" switch which writes configuration
values to app/etc/env.php. The "lock-config" switch writes it to
app/etc/config.php instead which can be shared between environments. 
For consistency, the "lock" switch has been renamed to "lock-env".

### Description

In most cases, configuration settings should be shared among all development, staging and live instances of a shop. To achieve this, you can use the same mechanism as with the `--lock` switch described in http://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-config-mgmt-set.html.

The only difference is that the configuration values will be stored in app/etc/config.php instead of app/etc/env.php. This file should usually be included in the VCS and thus shared between instances.

To store a config setting in the file via command line, the command would be as follows:

```
bin/magento config:set --lock-config <path> <value>
```

i.e. 

```
bin/magento config:set --lock-config general/store_information/name "Sample Store"
```

### Manual testing scenarios
1. Call `bin/magento config:set --lock-config general/store_information/name "Sample Store"` from the command line
2. Verify that the app/etc/config.php file contains a section as follows:

```
[...]
  'cache_types' => 
  array (
    'config' => 1,
    [...]
  ),
  'install' => 
  array (
    'date' => 'Wed, 08 Nov 2017 11:52:21 +0000',
  ),
  'system' => 
  array (
    'default' => 
    array (
      'general' => 
      array (
        'store_information' => 
        array (
          'name' => 'Sample Store',
        ),
      ),
    ),
  ),
);
```
3. Verify that the configuration setting "Store Name" in the section "General", group "Store Information" is filled with "Sample Store" and greyed out:

![grafik](https://user-images.githubusercontent.com/662059/32662987-ac0bffee-c62c-11e7-8641-97d5d9d3ac43.png)


### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

ishakhsuvarov committed:
Create CODE_OF_CONDUCT.md

localheinz committed:
Fix: Move GitHub-specific documents into .github

avstudnitz committed:
Add option "share" for shell command "config:set"

This is similar to the "lock" switch which writes configuration
values to app/etc/env.php. The "share" switch writes it to
app/etc/config.php instead which can be shared between environments.

avstudnitz committed:
Add unit tests for option "share" of shell command "config:set"

avstudnitz commented:
I am not totally happy with the naming of the option. Instead of `--share` I could think of `--lock-shared`. I think both are okay, but not perfect. I am open for suggestions.

avstudnitz committed:
Fix unit tests for option "share" of shell command "config:set"

fooman commented:
Thanks @avstudnitz - read your blog post on this and an addition like this totally makes sense. Agree that naming this is a tricky.

I would suggest the following: 
--lock-env (which is just an alias for --lock as to not brake anything)
--lock-config

avstudnitz commented:
You can find the blog post about this functionality which was mentioned by @fooman at https://www.integer-net.com/magento-2-2-configuration-via-code/.

jalogut commented:
Just referencing another PR with improvements in same area:

* #12410

avstudnitz committed:
12178 Refactor config:set command to use --lock-config instead of --share option

We are using Virtual Classes now as we have almost the same Processor
twice and we can avoid having duplicate code this way.

avstudnitz committed:
12178 Adjust unit tests

avstudnitz committed:
12178 Adjust integration tests

avstudnitz commented:
I adjusted the PR accordingly (now using Virtual Classes to avoid code duplication) and updated the description. Thanks for the suggestion @fooman.

avstudnitz committed:
12178 Fix integration tests

fooman commented:
@avstudnitz would you mind taking a look at this failure as well https://travis-ci.org/magento/magento2/jobs/306722542#L1031 which seems related.

avstudnitz committed:
12178 Fix integration tests

The "getOption" method was called more often than before, so
we had to adjust that.

avstudnitz committed:
12178 Code style optimizations to prevent CodeSniffer errors

avstudnitz committed:
12178 More code style optimizations to prevent CodeSniffer errors

avstudnitz commented:
@fooman Done.

osrecio committed:
Add Current Date to update_time Field Block and Cms

okorshenko commented on app/code/Magento/Config/Console/Command/ConfigSet/ProcessorFacade.php:
```diff
@@ -96,13 +97,13 @@ public function __construct(
      * @since 100.2.0
      */
-    public function process($path, $value, $scope, $scopeCode, $lock)
+    public function process($path, $value, $scope, $scopeCode, $lock, $lockTarget = ConfigFilePool::APP_ENV)
```
Unfortunately this backward incompatible change. Please follow the instruction here http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/#adding-parameters-in-public-methods

avstudnitz replied:
@okorshenko Can you explain to me why adding an optional parameter at the end should be backwards incompatible? I'd like to understand that.

okorshenko replied:
Hi @avstudnitz this class has annotation `@api`. Adding new parameter to the method will break SPI clients of this class

avstudnitz replied:
Changed. If you have a better name than `processWithLockTarget` for the new method please tell me.

RomaKis committed:
magento/magento2#8615: REST API unable to make requests with slash (/) in SKU

Commit:
Removed old code "I"

Old code "I" was not removed in initial commit for some reason.

ishakhsuvarov committed:
MAGETWO-85583: magento/magento2#12378: Regions list in Directory module for India #1007

 - Merge Pull Request magento-engcom/magento2ce#1007 from p-bystritsky/magento2:ISSUE-12378
 - Merged commits:
   1. 5c36f9888bc39efbfe479adb0e11d2d49f94d715
   2. c27a56a5d1527ffb3a66a375fc486675d716d54b

Commit:
Updated i18n

Added product codes to i18n

Commit:
Removed legacy codes from i18n

These codes have been renamed, so entries removed from i18n

ishakhsuvarov committed:
MAGETWO-85499: Magento Connect no longer exist #12633

 - Merge Pull Request magento/magento2#12633 from miguelbalparda/magento2:image-fix
 - Merged commits:
   1. ef1ffd5c61d50b44bcbeb663872dbb1236bb5f93
   2. cef5991721a1c85bc937b9d4d33fe79417ccc55e

Commit:
Fixed quotes around strings

Added quotes around strings in i18n

ishakhsuvarov committed:
MAGETWO-85579: 10814: Attribute repository resets sourceModel for new attributes. #1012

ishakhsuvarov committed:
MAGETWO-85549:  magento/magento2#12285: The option <var name="allowfullscreen">false</var> for mobile device don't work in product view page gallery  #1006

ishakhsuvarov committed:
MAGETWO-85546: magento/magento2#12259: Save and Duplicated product not working #983

ishakhsuvarov committed:
MAGETWO-85545: 8204: catalog:images:resize = getimagesize(): Read error! #1000

ishakhsuvarov committed:
MAGETWO-85499: Magento Connect no longer exist #12633

AVoskoboinikov committed:
MAGETWO-83370: Fix products distribution in shared catalogs

AVoskoboinikov committed:
Merge branch 'MAGETWO-81090' of https://github.com/magento-performance/magento2ce into MAGETWO-81090

p-bystritsky committed:
magento/magento2#12084: Product csv import > fail on round brackets in image filename

serhii-balko committed:
#11743: [GitHub] AbstractPdf - ZendException font is not set

Commit:
Merge pull request #1844 from magento-engcom/2.2-develop-prs

Merge pull request #1014 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop

Public Pull Requests
 - #1012 10814: Attribute repository resets sourceModel for new attributes. by @nmalevanec
- #1006 magento/magento2#12285: The option false for mobile device don't work in product -
view page gallery by @p-bystritsky
- #983 magento/magento2#12259: Save and Duplicated product not working by @p-bystritsky
- #1000 8204: catalog:images:resize = getimagesize(): Read error! by @RomaKis
magento/magento2#12633 Magento Connect no longer exist by @miguelbalparda

Fixed Public Issues
 - magento/magento2#10814 Attribute repository resets sourceModel for new attributes
 - magento/magento2#12285 The option false for mobile device don't work in product view page gallery
 - magento/magento2#12490 I can't disable full screen gallery on mobile on magento 2.2.1
 - magento/magento2#12259 Save and Duplicated product not working
 - magento/magento2#8204 catalog:images:resize = getimagesize(): Read error! in vendor/magento/module-catalog/Model/Product/Image.php on line 410 if an image is 0 bytes
 - magento/magento2#12632 Magento Connect no longer exist

andimov committed:
Merge branch '2.2' of https://github.com/magento/magento2ce into MAGETWO-85590

andimov committed:
MAGETWO-85050: Prepare codebase for 2.2.4

Commit:
magento/magento2#12063

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into RomaKis-11946

Commit:
MAGETWO-84370: 11946: Layer navigation showing wrong product count #12063

 - Merge Pull Request magento/magento2#12063 from RomaKis/magento2:11946
 - Merged commits:
   1. 0c82649c1f6e9151a5503782bdf874e3d4606fa8
   2. 1ef474cb75650e68f4ea7f13e9704a237eab9fe7
   3. e195dd6075dfa54392c7d444bdab3eb63596ff51

duhon committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

nmalevanec committed:
6965: magento/magento2#6965: \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate

RomaKis committed:
magento/magento2#12656: Checkout: Whitespace in front of coupon code causes "Coupon code is not valid"

serhii-balko committed:
#11743: [GitHub] AbstractPdf - ZendException font is not set

serhii-balko committed:
#11743: [GitHub] AbstractPdf - ZendException font is not set

ishakhsuvarov committed:
MAGETWO-85580: Fixes #12660 invalid parameter configuration provided for  argument #12661

 - Merge Pull Request magento/magento2#12661 from lcbrq/magento2:2.2-develop
 - Merged commits:
   1. c7abab1d2351d8c6dad59d5bb153f23924ae2e6d

Commit:
Reverted change for product code "O"

Product code "O" changed in error. This reverts the change.

duhon committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

RomaKis committed:
magento/magento2#12667: Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility "Not Visible Individually"

AVoskoboinikov committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

ishakhsuvarov committed:
MAGETWO-85583: magento/magento2#12378: Regions list in Directory module for India #1007

ishakhsuvarov committed:
MAGETWO-85580: Fixes #12660 invalid parameter configuration provided for  argument #12661

ishakhsuvarov committed:
MAGETWO-85534: #8647: [GitHub] Order of how arguments are merged in multiple di.xml-… #995

ishakhsuvarov committed:
MAGETWO-84370: 11946: Layer navigation showing wrong product count #12063

p-bystritsky committed:
magento/magento2#10734: Magento 2 is not showing Popular Search Terms [backport]

ishakhsuvarov committed:
MAGETWO-85610: magento/magento2#12656: Checkout: Whitespace in front of coupon code causes 'Coupon code is not valid' #1021

 - Merge Pull Request magento-engcom/magento2ce#1021 from RomaKis/magento2:12656
 - Merged commits:
   1. 7982b300b4d7e20f60f0873c246739e02adbd390

ishakhsuvarov committed:
Merge pull request #1848 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-85583: magento/magento2#12378: Regions list in Directory module for India #1007
 - MAGETWO-85580: Fixes #12660 invalid parameter configuration provided for argument #12661
 - MAGETWO-85534: #8647: [GitHub] Order of how arguments are merged in multiple di.xml-… #995
 - MAGETWO-84370: 11946: Layer navigation showing wrong product count #12063
Fixed issues:
 - magento/magento2#12378 Regions list in Directory module for India
 - magento/magento2#12452 ACL permissions issue
 - magento/magento2#12660 Invalid parameter configuration provided for $block argument upon no ACL permissions to the block
 - magento/magento2#8647 Order of how arguments are merged in multiple di.xml-files causes unexpected results
 - magento/magento2#11946 Layer navigation showing wrong product count

ishakhsuvarov committed:
MAGETWO-85629:  magento/magento2#12084: Product csv import > fail on round brackets in image filename #1017

 - Merge Pull Request magento-engcom/magento2ce#1017 from p-bystritsky/magento2:ISSUE-12084
 - Merged commits:
   1. c27282c5c7c67fe5ea2c8e03fa99a9b029c54907

Commit:
magento/magento2#12666

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into gwharton-2.2-develop-gw-dhl

Commit:
MAGETWO-85586: Fix incorrect DHL Product codes #12666

 - Merge Pull Request magento/magento2#12666 from gwharton/magento2:2.2-develop-gw-dhl
 - Merged commits:
   1. 4d5ea55540002dcdfb47784943fab60c5f5dde07
   2. 30a426bd996a5bb95e2bcf2a7a99ed02774a9171
   3. a6abc596a1e5a6c901cc721393000e8e25e6dff5
   4. b422f2c8ad83127a725d12336c5bd287c745128d
   5. 2ba87c689ef2a8d356ddd5a986a9fa2abd9e13ce
   6. f99ae4ddf32873bed7e6e1cb5a2237a84fe4ff16
   7. cc10dc1d4f6ca7e2d88fd05122facacc0d0cf5ae
   8. b9d01bf1e059ad5c61beae66014af9f8a0ee2a21

Commit:
MAGETWO-85634: magento/magento2#12667: Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility 'Not Visible Individually' #1023

 - Merge Pull Request magento-engcom/magento2ce#1023 from RomaKis/magento2:12667
 - Merged commits:
   1. 585803acd949ac5f1be4297ec93144b4e1de74de

AVoskoboinikov committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

YevSent committed:
MAGETWO-81395: Provide possibility to customize error messages from payment methods

 - Added interface to payment gateway infrastructure to provide error messages customization
 - Implemented error messages mapping for Braintree payment method
 - Refactored affected tests
 - Added messages localization

AVoskoboinikov committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

serhii-balko committed:
#11743: [GitHub] AbstractPdf - ZendException font is not set

AVoskoboinikov committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

Commit:
updated code following code review & backward compatible development guide

p-bystritsky committed:
magento/magento2#10734: Magento 2 is not showing Popular Search Terms [backport]

Commit:
updated code following code review & backward compatible development guide

ishakhsuvarov committed:
MAGETWO-85610: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85610-magento-engcom-magento2ce-1021

AVoskoboinikov committed:
MAGETWO-83365: Stabilize Frontend Pool for multi thread run

Commit:
fixed code styles

dmanners committed:
MAGETWO-85643: Add price calculation improvement for product option value price #11563

 - Merge Pull Request magento/magento2#11563 from marinagociu/magento2:fix-custom-option-price
 - Merged commits:
   1. 33aa293e2337101aac6250a9bdf9c93bf8af3377
   2. 7ff22bf30987dc0902d10eae1b11f797a3b9ff75

p-bystritsky committed:
magento/magento2#10734: Magento 2 is not showing Popular Search Terms [backport]

nmalevanec committed:
6965: magento/magento2#6965: \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate

Commit:
add nullable in comment & code cleanup

Commit:
MAGETWO-85637: [Backport] magento/magento2#10743: Magento 2 is not showing Popular Search Terms #1024

 - Merge Pull Request magento-engcom/magento2ce#1024 from p-bystritsky/magento2:ISSUE-10734
 - Merged commits:
   1. 1f77aaad920551ae0309a9964989213f6c265e4f
   2. e340347310cfd1c2e91e79f07d21291de7a55348
   3. 43ede3ca2718c3c3bb9242e99a8bd121680ecf6e

Commit:
code cleanup

ishakhsuvarov committed:
MAGETWO-85659: 6965:  \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate #1022

 - Merge Pull Request magento-engcom/magento2ce#1022 from nmalevanec/magento2:6965
 - Merged commits:
   1. 2c4fa0771676b562f84de41069d64aff648172ba
   2. ed5ffedd4cd28492f299e5d059414dd3d2540b8e

ishakhsuvarov committed:
MAGETWO-85660: #11743: [GitHub] AbstractPdf - ZendException font is not set #1016

 - Merge Pull Request magento-engcom/magento2ce#1016 from serhii-balko/magento2:github-11743
 - Merged commits:
   1. 2629726dd4754b5fd50eebb4860794b7931fdd88
   2. b326714be1661f917b23fea90576512b9c631f99
   3. 5f4363b243fa0c297a8be2cab4c00f0d806b14c1
   4. 9ce4730da51d91f1a1925f41b9ae6d573d9e96bd

andimov committed:
Merge branch '2.2-develop' of https://github.com/magento/magento2ce into MAGETWO-85590

Commit:
MAGETWO-85643: Add price calculation improvement for product option value price #11563

Commit:
MAGETWO-85637: [Backport] magento/magento2#10743: Magento 2 is not showing Popular Search Terms #1024

Commit:
MAGETWO-85634: magento/magento2#12667: Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility "Not Visible Individually" #1023

Commit:
MAGETWO-85629:  magento/magento2#12084: Product csv import > fail on round brackets in image filename #1017

Commit:
MAGETWO-85610: magento/magento2#12656: Checkout: Whitespace in front of coupon code causes "Coupon code is not valid" #1021

Commit:
MAGETWO-85586: Fix incorrect DHL Product codes #12666

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into 8830

omiroshnichenko committed:
magento/magento2#8830: Can`t delete row in dynamicRows component

Commit:
merge magento/2.2-develop into magento-performance/MAGETWO-81090

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into quisse-2.2

omiroshnichenko committed:
MAGETWO-85672: Add customer login url from Customer Url model to checkout config so … #12630

 - Merge Pull Request magento/magento2#12630 from quisse/magento2:2.2-develop
 - Merged commits:
   1. 6f7219f4d2a6cb43cb3b7e8805a096d6df63ca44
   2. 2602e7ec18fd2f0bed5fefeee91a52b0fbde160e
   3. 115ba8ef9d739ee131fccf5baff6d3a8b9d4b192
   4. 97f77e64e7db6c04958632906ae9519590918414
   5. bb3932cfdb8985fe8c1e752ab99414bdffc47ebf
   6. 8986107dfe5dadb582fc11113287cc933b06cf9e
   7. 5c87fb020fb2317eeaef1dc76132b3bd272b05ec
   8. 4350858fe93f76ecebcb560ba746004f66570192

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into experius-2.2-patch-customer-accountmanagement-is-confirmation-required

Commit:
Merge pull request #1851 from magento-engcom/2.2-develop-prs

Public Pull Requests
 - magento/magento2#11563 Add price calculation improvement for product option value price by @marinagociu
 - magento-engcom#1024 [Backport] magento/magento2#10734: Magento 2 is not showing Popular Search Terms by @p-bystritsky
 - magento-engcom#1023 magento/magento2#12667: Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility "Not Visible Individually" by @RomaKis
 - magento-engcom#1017 magento/magento2#12084: Product csv import > fail on round brackets in image filename by @p-bystritsky
 - magento-engcom#1021 magento/magento2#12656: Checkout: Whitespace in front of coupon code causes "Coupon code is not valid" by @RomaKis
 - magento/magento2#12666 Fix incorrect DHL Product codes by @gwharton

Fixed Public Issues
 - magento/magento2#5774 Tier price and custom options give bad results
 - magento/magento2#10743 Magento 2 is not showing Popular Search Terms
 - magento/magento2#12667 Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility "Not Visible Individually"
 - magento/magento2#12084 Product csv import > fail on round brackets in image filename
 - magento/magento2#12656 Checkout: Whitespace in front of coupon code causes "Coupon code is not valid"

ishakhsuvarov committed:
Deny access to .github directory

ishakhsuvarov committed:
MAGETWO-85694: [2.2 Backport] Create CODE_OF_CONDUCT.md #12723

 - Merge Pull Request magento/magento2#12723 from ishakhsuvarov/magento2:github-specific-documents
 - Merged commits:
   1. f69dd0fd83b067a84faddebd378837568866c4bc
   2. 104faf3cd9d4353cfa7f6dc86abb828eebc412cb
   3. 3d2e1eb5e9ddf20b6c4e78bedfea2d2403b62ff3

ishakhsuvarov committed:
MAGETWO-85694: [2.2 Backport] Create CODE_OF_CONDUCT.md #12723

ishakhsuvarov committed:
Merge pull request #1853 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-85694: [2.2 Backport] Create CODE_OF_CONDUCT.md #12723

orlangur committed:
MAGETWO-85698: Remove deprecation without alternative #11070

 - Merge Pull Request magento/magento2#11070 from schmengler/magento2:undeprecate-cart
 - Merged commits:
   1. e5479b4e7b87b8ad22dc1ca8e4c33556999da5f2

slopukhov committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-81090

RomaKis committed:
magento/magento2#8615: REST API unable to make requests with slash (/) in SKU

vzabaznov committed:
Merge pull request #1845 from magento-helix/MAGETWO-85590

[Helix] merge 2.2 into 2.2-develop

Commit:
magento/magento2
 - updated the "currency-addon-symbol__width" variable in "app/design/adminhtml/Magento/backend/Magento_ConfigurableProduct".

Commit:
Merge branch '2.2-develop' of https://github.com/magento/magento2 into 2.2-develop

Commit:
magento/magento2#12713: Currency symbol overlaps entered attribute option's price while creating Configurable Product
- updated the "currency-addon-symbol__width" variable in "app/design/adminhtml/Magento/backend/Magento_ConfigurableProduct".

AVoskoboinikov committed:
MAGETWO-83659: Enable metrics validation for PAT

AVoskoboinikov committed:
Merge branch '2.2-develop' of https://github.com/magento/magento2ce into MAGETWO-81090

ihor-sviziev committed:
magento/magento2#12206: Tracking link returns 404 page in admin panel

virtual97 committed:
magento/magento2#12374: Model hasDataChanges always true. -Added false for collection items.

virtual97 committed:
Merge remote-tracking branch 'remotes/origin/2.2-develop' into issues/12374

dmanners committed:
magento/magento2#7241: No option to start with blank option for prefix and suffix in checkout.
 - replace usage of empty string for optional value with space so that the knockout.js picks up the optional values

dzianis-yurevich committed:
#6916 Fix notice during Update Bundle Product without changes in bundle items

dmanners committed:
MAGETWO-82635: #7241: Always add empty option for prefix and/or suffix if optional #11462

 - Merge Pull Request magento/magento2#11462 from avstudnitz/magento2:7241
 - Merged commits:
   1. 11e38f68a9956cbdc1d7c48284bc712729f43a90
   2. 7800e5dd8aac704da3f2b1799baadf2a01269c61
   3. 252529f0819d2792982452ed9d929d5fb526996d
   4. 519b805c2853f895d804c49e08bd7a23b108c38c
   5. eb79b38beb6b79411fe95d30e324d7d094ab5874
   6. fce3afe6390be6c0eb2d5fcdb2f76108d6a44e77
   7. 3e38118c1e7f92e77963a6ae4f42e66520e6f58f

xpoback committed:
magento/magento2#12719: Use full name in welcome message

AndrewMun committed:
magento/magento2#11953: Product configuration creator does not warn about invalid SKUs
- Added validation for SKU field inside "Configurations" tab

zamoroka committed:
magento/magento2#11953: Product configuration creator does not warn about invalid SKUs
 - Make sku field as required
 - Length of sku gets from Sku model

AndrewMun committed:
magento/magento2#6113: Validate range-words in Form component (UI Component)
- Fixed regexp pattern for range-words validation

strelldev committed:
#9453 - ported down c2e5d77a9516c8305585e819c2f0a0629648cc14
 - MAGETWO-70322: [2.1.x][GitHub] SID in URL even if disabled #9453

hannassy committed:
magento/magento2#9720 Menu item dependencies (dependsOnModule, dependsOnConfig) are broken

serhii-balko committed:
#12712: [GitHub] Latest Google Chrome Browser issue with duplicate id="email"

Styopchik committed:
magento/magento2#12439 Newsletter subscription success email not sent after confirmation
- added sendSuccessMessage

ihor-sviziev committed:
magento/magento2#12206: Tracking link returns 404 page in admin panel

Cover changes with integration test

omiroshnichenko committed:
magento/magento2#8830: Can`t delete row in dynamicRows component
- fix template
- fix sort order for dropdown attribute

Mobecls committed:
magento/magento2#5035 

Cannot subscribe to events with a number in name

Mobecls committed:
magento/magento2#5035

Cannot subscribe to events with a number in name

StasKozar committed:
magento/magento2#12715 Storefront Back to Sign in button does not work
- add Back to Sign in url

StasKozar committed:
magento/magento2#12715 Storefront Back to Sign in button does not work
- add Back to Sign in url

AndrewMun committed:
magento/magento2#6113: Validate range-words in Form component (UI Component)
- Changed range-words validation logic

Commit:
magento/magento2#12699: fixed attribute loading for different entities in AbstractEntity

omiroshnichenko committed:
magento/magento2#6113: Validate range-words in Form component (UI Component)
- add test

vzabaznov committed:
Merge pull request #1850 from magento-performance/MAGETWO-81090

[PERFORMANCE] PAT Build improvement

Commit:
MAGETWO-85287: 8615: REST API unable to make requests with slash (/) in SKU #949

 - Merge Pull Request magento-engcom/magento2ce#949 from RomaKis/magento2:8615
 - Merged commits:
   1. 377668493512c25d6ba94c2fdff1a9f6a35e3530
   2. 8870cf2a0943bcc7c85e44782284baeaf2ab39b9

Commit:
MAGETWO-85698: Remove deprecation without alternative #11070

Commit:
MAGETWO-85287: 8615: REST API unable to make requests with slash (/) in SKU #949

Commit:
MAGETWO-85741: Fix Back to Sign in url on confirmation form #12759

 - Merge Pull Request magento/magento2#12759 from StasKozar/magento2:2.2-develop
 - Merged commits:
   1. 946382cc9063ad90fd14bc3890d2230ba606ee4c
   2. faee0fc0cef419b867bdaa83fc47062ffeb6b1fc

Commit:
Merge pull request #1856 from magento-engcom/2.2-develop-prs

Public Pull Requests
 - magento/magento2#11070 Remove deprecation without alternative by @schmengler
 - magento-engcom#949 8615: REST API unable to make requests with slash (/) in SKU by @RomaKis

Fixed Public Issues
 - magento/magento2#10133 Please add your expectations for @deprecated annotations
 - magento/magento2#8615 REST API unable to make requests with slash (/) in SKU

Commit:
:pencil2: Fix typo

Fix typo in SINGLE_PRODUCT_LAYOUT_HANLDE to SINGLE_PRODUCT_LAYOUT_HANDLE

orlangur committed:
MAGETWO-85770: magento/magento2#5035 Cannot subscribe to events with a number in name #12758

 - Merge Pull Request magento/magento2#12758 from Mobecls/magento2:bugfix
 - Merged commits:
   1. 937b26d3cc23ce86df6db204765b10f4ccf7ea5a
   2. 85fa67170f413ea5f509c7f1e0cf870cae559bb3

Commit:
MAGETWO-85713: magento/magento2#12206: Tracking link returns 404 page in admin panel #12732

 - Merge Pull Request magento/magento2#12732 from ihor-sviziev/magento2:12206-tracking-link-returns-404-page-in-admin
 - Merged commits:
   1. ecf27becd6bc99842527153f45e0e7fc91c95624
   2. 5e4bf42b29de485cf19b119e32e52effe5ce1820

dmanners committed:
MAGETWO-85777: magento/magento2#11953: Product configuration creator does not warn about invalid SKUs #12737

 - Merge Pull Request magento/magento2#12737 from zamoroka/magento2:issues/11953
 - Merged commits:
   1. 5cf0d3480fcc302f922fce12a588afd5e5771491
   2. 7418f1ac607c68003c2f4122f2b1bf07a35c044e

ishakhsuvarov committed:
MAGETWO-85779: magento/magento2#9720 Menu item dependencies (dependsOnModule, depend… #12747

 - Merge Pull Request magento/magento2#12747 from hannassy/magento2:2.2-develop#9720
 - Merged commits:
   1. ddf6c9fc2f7210d438d658d950e7f3c1c55f6bf3

ishakhsuvarov committed:
MAGETWO-85780: #9453 - ported down c2e5d77a9516c8305585e819c2f0a0629648cc14 #12743

 - Merge Pull Request magento/magento2#12743 from strell/magento2:2.2-develop-9453
 - Merged commits:
   1. 38720ea0920622b8b66e2b576ea7d630ff9294c1

ishakhsuvarov committed:
MAGETWO-85779: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85779-magento-magento2-12747

ishakhsuvarov committed:
MAGETWO-85780: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85780-magento-magento2-12743

Commit:
:ok_hand: Deprecate old constant

Deprecate old constant with typo and introduce new correct spelled one.

dmanners committed:
MAGETWO-85773: Fix typo in SINGLE_PRODUCT_LAYOUT_HANLDE #12786

 - Merge Pull Request magento/magento2#12786 from aschrammel/magento2:bug/handle-typo
 - Merged commits:
   1. 935dd8d294a5f24417dae8df3b1425e1113a76e9
   2. deffe81b0a99aada6f6121e5b70c10d4ea471e0d

miguelbalparda committed:
MAGETWO-85782: 12713 #12730

 - Merge Pull Request magento/magento2#12730 from EfremovaVI/magento2:12713
 - Merged commits:
   1. 370e526de6b6f67b5f07dea5bfbc91884f9311ea
   2. 5e43f84815c8a00f4216242fc9e3ee2e442f3552
   3. ff7029f231b4289ced181485708e35fca0ad6a42

miguelbalparda committed:
MAGETWO-85783: magento/magento2#12439 Newsletter subscription success email not sent… #12751

 - Merge Pull Request magento/magento2#12751 from Styopchik/magento2:#12439
 - Merged commits:
   1. c4e4168cdeb349bd45fcb609970ab1219231c8f7

ishakhsuvarov committed:
MAGETWO-85784: magento/magento2#6113: Validate range-words in Form component (UI Component) #12739

 - Merge Pull Request magento/magento2#12739 from zamoroka/magento2:issue/6113
 - Merged commits:
   1. 465e14bdee949bf8c92ed1640119930e0cce4091
   2. 8bc807fb097ff2ef8f45ff756d69e27f563cf388
   3. e9dcb03d2430e0b4289502c3550c0e24fd945e3b

ishakhsuvarov committed:
MAGETWO-85784: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85784-magento-magento2-12739

ishakhsuvarov committed:
MAGETWO-85785: magento/magento2#12719: Use full name in welcome message #12738

 - Merge Pull Request magento/magento2#12738 from xpoback/magento2:2.2-develop
 - Merged commits:
   1. 6b9c39cadfe86a62a5b0c4b13470431f0409c651

ishakhsuvarov committed:
MAGETWO-85785: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85785-magento-magento2-12738

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into experius-2.2-patch-customer-accountmanagement-is-confirmation-required

omiroshnichenko committed:
MAGETWO-83026: [BUGFIX] Made method public so a plugin is possible. #11878

 - Merge Pull Request magento/magento2#11878 from experius/magento2:experius-2.2-patch-customer-accountmanagement-is-confirmation-required
 - Merged commits:
   1. 8573467f9f40c76d362b5992e70de9f7cd8782ed
   2. 7ad5db6d7404d6e04f3b91e081b379e048b519e2
   3. 6227a14c301604008e4b62e061fdc482f6b66371
   4. 0a545ebe2f0c933ea4a9d30f4cbd94bbb351e4d2
   5. 880dd9fa672267a3257c7631fd7d73781b79f06f

Commit:
magento/magento2#12699: new attribute should be private

Commit:
magento/magento2#12767

Commit:
MAGETWO-85772: magento/magento2#12699: Multiselect Attribute is not saved #12767

 - Merge Pull Request magento/magento2#12767 from awarche/magento2:2.2-develop
 - Merged commits:
   1. 39098d893e7514c4ac6ebbbd10675cfb06289f29
   2. 49ae5fa7c70fa896180131a4c30f08624f2e5821
   3. 5c3067093a9a8bfa66566adc200b45a6192f55c5

nicka101 committed:
Stop the profiler when returning early

Magento\Eav\Model\Config::getAttribute doesn't stop the Profiler when returning early, and thus incorrectly reports run time in these cases. 
Fix it by stopping the profiler before return

veloraven committed:
Merge branch 'MAGETWO-84464' into Chaika_PR_2.2.4

dmanners committed:
MAGETWO-85833: Stop the profiler when returning early in \Magento\Eav\Model\Config::getAttribute #12810

 - Merge Pull Request magento/magento2#12810 from nicka101/magento2:patch-1
 - Merged commits:
   1. 5f1b8461045779966bb5e70954d4ff96dd2447d9

avstudnitz committed:
7241 Restore old @api method "process" and add a new method

This new method contains the new parameter and will be called from
the old method.

avstudnitz committed:
7241 Restore old @api method "process" and add a new method

This new method contains the new parameter and will be called from
the old method.

dmanners committed:
magento/magento2#5188: Revert the loading of xml from the factory back into the phpstorm specific file

dmanners committed:
magento/magento2#5188: Refactor the basic file setup into it's own method

Commit:
Fix PhpDoc to show correct parameter types

dmanners committed:
magento/magento2#5188: Remove the no longer needed xml catalog domfactory class

dmanners committed:
magento/magento2#5188: Update DomDocumentFactoryTest to check that create returns the expected type of object

dmanners committed:
MAGETWO-83120:  Fix error when generating urn catalog for empty misc.xml #11686

 - Merge Pull Request magento/magento2#11686 from tdgroot/magento2:2.2-develop-fix_empty_idea_misc_xml
 - Merged commits:
   1. 177e540938631fbb749dcc028d4f8f864e290b4e
   2. 0d652139889d882dd6646741070e12d04ebd8a5b
   3. a781bf6f75c74a6e871f1640ae679b18967ac000
   4. dcc480d15c9f0c1aca77974d5cd89a972f1d32dd
   5. 5d4e1c29ca7d0a62a3465dd7b98a5c30d1410845
   6. b552dd1be35436e05065d8269dabe6cebfaaafcd
   7. b85b91e4fa5c41fe337accd4709b184d7811cd65

tzyganu committed:
mark method retrieveAttributeSetIdFromSearchCriteria as deprecated

mzeis committed:
MAGETWO-85875: #11936:required attribute set id filter on attribute group repository getList #12105

 - Merge Pull Request magento/magento2#12105 from tzyganu/magento2:M2-11936
 - Merged commits:
   1. 433c3cd97c327b6130fa34ad2aad15659facafe3
   2. 7be92a872300f949d349e442c3d8e1045609d8b1

Commit:
Merge pull request #1867 from magento-chaika/Chaika_PR_2.2.4

Chaika_PR_2.2.4

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

scribam committed:
functional.suite.dist.yml: use the value provided by the .env file for the backend_name configuration

mslabko committed:
MAGETWO-82888: [Performance] Product Price indexer stuck during reindexing

Commit:
MAGETWO-85782: 12713 #12730

Commit:
MAGETWO-85780: #9453 - ported down c2e5d77a9516c8305585e819c2f0a0629648cc14 #12743

Commit:
MAGETWO-85779: magento/magento2#9720 Menu item dependencies (dependsOnModule, depend… #12747

Commit:
MAGETWO-85773: Fix typo in SINGLE_PRODUCT_LAYOUT_HANLDE #12786

Commit:
MAGETWO-85772: magento/magento2#12699: Multiselect Attribute is not saved #12767

mzeis committed:
MAGETWO-85502: #12625: Add Current Date to update_time Field for Block and Pages #12636

 - Merge Pull Request magento/magento2#12636 from osrecio/magento2:PR#12625_2.2
 - Merged commits:
   1. 22f454dd020719686c558a36d67df6dc13506c3f

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12730: 12713 (by @EfremovaVI)
 - magento/magento2#12743: #9453 - ported down c2e5d77a9516c8305585e819c2f0a0629648cc14 (by @strell)
 - magento/magento2#12747: magento/magento2#9720 Menu item dependencies (dependsOnModule, depend… (by @hannassy)
 - magento/magento2#12786: Fix typo in SINGLE_PRODUCT_LAYOUT_HANLDE (by @aschrammel)
 - magento/magento2#12767: magento/magento2#12699: Multiselect Attribute is not saved (by @awarche)


Fixed GitHub Issues:
 - magento/magento2#12713: Currency symbol overlaps entered attribute option's price while creating Configurable Product (reported by @alena-marchenko) has been fixed in magento/magento2#12730 by @EfremovaVI in 2.2-develop branch
   Related commits:
     1. 370e526de6b6f67b5f07dea5bfbc91884f9311ea
     2. 5e43f84815c8a00f4216242fc9e3ee2e442f3552
     3. ff7029f231b4289ced181485708e35fca0ad6a42

 - magento/magento2#9453: Reopened: '?SID' in URL even if disabled (reported by @hostep) has been fixed in magento/magento2#12743 by @strell in 2.2-develop branch
   Related commits:
     1. 38720ea0920622b8b66e2b576ea7d630ff9294c1

 - magento/magento2#9720: Menu item dependencies (dependsOnModule, dependsOnConfig) are broken (reported by @mam08ixo) has been fixed in magento/magento2#12747 by @hannassy in 2.2-develop branch
   Related commits:
     1. ddf6c9fc2f7210d438d658d950e7f3c1c55f6bf3

Commit:
MAGETWO-85872: Fix PhpDoc to show correct parameter types #12826

 - Merge Pull Request magento/magento2#12826 from FreekVandeursen/magento2:phpdoc/abstractcollection
 - Merged commits:
   1. aa5ed165f91e71b979b033cab21cfcb3e867f80c

Commit:
MAGETWO-85833: Stop the profiler when returning early in \Magento\Eav\Model\Config::getAttribute #12810

Commit:
MAGETWO-85785: magento/magento2#12719: Use full name in welcome message #12738

Commit:
MAGETWO-85784: magento/magento2#6113: Validate range-words in Form component (UI Component) #12739

Commit:
MAGETWO-85770: magento/magento2#5035 Cannot subscribe to events with a number in name #12758

Commit:
MAGETWO-85741: Fix Back to Sign in url on confirmation form #12759

Commit:
MAGETWO-85713: magento/magento2#12206: Tracking link returns 404 page in admin panel #12732

Commit:
MAGETWO-85672: Add customer login url from Customer Url model to checkout config so … #12630

Commit:
MAGETWO-85659: 6965:  \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate #1022

Commit:
MAGETWO-85872: Fix PhpDoc to show correct parameter types #12826

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12810: Stop the profiler when returning early in \Magento\Eav\Model\Config::getAttribute (by @nicka101)
 - magento/magento2#12738: magento/magento2#12719: Use full name in welcome message (by @xpoback)
 - magento/magento2#12739: magento/magento2#6113: Validate range-words in Form component (UI Component) (by @zamoroka)
 - magento/magento2#12758: magento/magento2#5035 Cannot subscribe to events with a number in name (by @Mobecls)
 - magento/magento2#12759: Fix Back to Sign in url on confirmation form (by @StasKozar)
 - magento/magento2#12732: Fix issue when tracking link returns 404 page in admin panel (by @ihor-sviziev)
 - magento/magento2#12630: Add customer login url from Customer Url model to checkout config so … (by @quisse)
 - magento-engcom/magento2ce#1022: 6965:  \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate (by @nmalevanec)
 - magento/magento2#12826: Fix PhpDoc to show correct parameter types (by @FreekVandeursen)


Fixed GitHub Issues:
 - magento/magento2#12719: Welcome message is shown with customer's first and last names after confirming account (reported by @alena-marchenko) has been fixed in magento/magento2#12738 by @xpoback in 2.2-develop branch
   Related commits:
     1. 6b9c39cadfe86a62a5b0c4b13470431f0409c651

 - magento/magento2#6113: Validate range-words in Form component (UI Component) (reported by @robinhuy) has been fixed in magento/magento2#12739 by @zamoroka in 2.2-develop branch
   Related commits:
     1. 465e14bdee949bf8c92ed1640119930e0cce4091
     2. 8bc807fb097ff2ef8f45ff756d69e27f563cf388
     3. e9dcb03d2430e0b4289502c3550c0e24fd945e3b

 - magento/magento2#5035: I can not to subscribe on change of all sections in Stores ->Configuration using event admin_system_config_changed_section (reported by @katesuykovskaya) has been fixed in magento/magento2#12758 by @Mobecls in 2.2-develop branch
   Related commits:
     1. 937b26d3cc23ce86df6db204765b10f4ccf7ea5a
     2. 85fa67170f413ea5f509c7f1e0cf870cae559bb3

 - magento/magento2#12715: Storefront Back to Sign in button does not work as expected (reported by @alena-marchenko) has been fixed in magento/magento2#12759 by @StasKozar in 2.2-develop branch
   Related commits:
     1. 946382cc9063ad90fd14bc3890d2230ba606ee4c
     2. faee0fc0cef419b867bdaa83fc47062ffeb6b1fc

 - magento/magento2#12206: Tracking link returns 404 page in admin panel (reported by @catstantin) has been fixed in magento/magento2#12732 by @ihor-sviziev in 2.2-develop branch
   Related commits:
     1. ecf27becd6bc99842527153f45e0e7fc91c95624
     2. 5e4bf42b29de485cf19b119e32e52effe5ce1820

 - magento/magento2#12627: Referer is not added to login url in checkout config (reported by @quisse) has been fixed in magento/magento2#12630 by @quisse in 2.2-develop branch
   Related commits:
     1. 6f7219f4d2a6cb43cb3b7e8805a096d6df63ca44
     2. 2602e7ec18fd2f0bed5fefeee91a52b0fbde160e
     3. 115ba8ef9d739ee131fccf5baff6d3a8b9d4b192
     4. 97f77e64e7db6c04958632906ae9519590918414
     5. bb3932cfdb8985fe8c1e752ab99414bdffc47ebf
     6. 8986107dfe5dadb582fc11113287cc933b06cf9e
     7. 5c87fb020fb2317eeaef1dc76132b3bd272b05ec
     8. 4350858fe93f76ecebcb560ba746004f66570192

 - magento/magento2#6965: \Magento\Directory\Model\PriceCurrency::format() fails without conversion rate (reported by @Vinai) has been fixed in magento-engcom/magento2ce#1022 by @nmalevanec in 2.2-develop branch
   Related commits:
     1. 2c4fa0771676b562f84de41069d64aff648172ba
     2. ed5ffedd4cd28492f299e5d059414dd3d2540b8e

Commit:
MAGETWO-85875: #11936:required attribute set id filter on attribute group repository getList #12105

Commit:
MAGETWO-85783: magento/magento2#12439 Newsletter subscription success email not sent… #12751

Commit:
MAGETWO-85777: magento/magento2#11953: Product configuration creator does not warn about invalid SKUs #12737

Commit:
MAGETWO-85660: #11743: [GitHub] AbstractPdf - ZendException font is not set #1016

Commit:
MAGETWO-85502: #12625: Add Current Date to update_time Field for Block and Pages #12636

Commit:
MAGETWO-83120:  Fix error when generating urn catalog for empty misc.xml #11686

Commit:
MAGETWO-83026: [BUGFIX] Made method public so a plugin is possible. #11878

Commit:
MAGETWO-82635: #7241: Always add empty option for prefix and/or suffix if optional #11462

Commit:
Throw ValidationException for invalid xml

This change ensures the offending file path is output in the error
report.

YevSent committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-81395

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into 8830

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12105: #11936:required attribute set id filter on attribute group repository getList (by @tzyganu)
 - magento/magento2#12751: magento/magento2#12439 Newsletter subscription success email not sent… (by @Styopchik)
 - magento/magento2#12737: magento/magento2#11953: Product configuration creator does not warn about invalid SKUs (by @zamoroka)
 - magento-engcom/magento2ce#1016: #11743: [GitHub] AbstractPdf - ZendException font is not set (by @serhii-balko)
 - magento/magento2#12636: #12625: Add Current Date to update_time Field for Block and Pages (by @osrecio)
 - magento/magento2#11686:  Fix error when generating urn catalog for empty misc.xml (by @tdgroot)
 - magento/magento2#11878: [BUGFIX] Made method public so a plugin is possible. (by @dheesbeen)
 - magento/magento2#11462: #7241: Always add empty option for prefix and/or suffix if optional (by @avstudnitz)


Fixed GitHub Issues:
 - magento/magento2#11936: required attribute set id filter on attribute group repository getList (reported by @tzyganu) has been fixed in magento/magento2#12105 by @tzyganu in 2.2-develop branch
   Related commits:
     1. 433c3cd97c327b6130fa34ad2aad15659facafe3
     2. 7be92a872300f949d349e442c3d8e1045609d8b1

 - magento/magento2#12439: Newsletter subscription success email not sent after confirmation (reported by @nfourteen) has been fixed in magento/magento2#12751 by @Styopchik in 2.2-develop branch
   Related commits:
     1. c4e4168cdeb349bd45fcb609970ab1219231c8f7

 - magento/magento2#11953: Product configuration creator does not warn about invalid SKUs (reported by @JanisE) has been fixed in magento/magento2#12737 by @zamoroka in 2.2-develop branch
   Related commits:
     1. 5cf0d3480fcc302f922fce12a588afd5e5771491
     2. 7418f1ac607c68003c2f4122f2b1bf07a35c044e

 - magento/magento2#11743: AbstractPdf - ZendException font is not set (reported by @jchenevey) has been fixed in magento-engcom/magento2ce#1016 by @serhii-balko in 2.2-develop branch
   Related commits:
     1. 2629726dd4754b5fd50eebb4860794b7931fdd88
     2. b326714be1661f917b23fea90576512b9c631f99
     3. 5f4363b243fa0c297a8be2cab4c00f0d806b14c1
     4. 9ce4730da51d91f1a1925f41b9ae6d573d9e96bd

 - magento/magento2#12625: when saving a page in magento 2.2.1, 'Modified' date field is not getting updated (reported by @mahmood537) has been fixed in magento/magento2#12636 by @osrecio in 2.2-develop branch
   Related commits:
     1. 22f454dd020719686c558a36d67df6dc13506c3f

 - magento/magento2#5188: Error generating URN-catalog when blank one exists (reported by @JamesonNetworks) has been fixed in magento/magento2#11686 by @tdgroot in 2.2-develop branch
   Related commits:
     1. 177e540938631fbb749dcc028d4f8f864e290b4e
     2. 0d652139889d882dd6646741070e12d04ebd8a5b
     3. a781bf6f75c74a6e871f1640ae679b18967ac000
     4. dcc480d15c9f0c1aca77974d5cd89a972f1d32dd
     5. 5d4e1c29ca7d0a62a3465dd7b98a5c30d1410845
     6. b552dd1be35436e05065d8269dabe6cebfaaafcd
     7. b85b91e4fa5c41fe337accd4709b184d7811cd65

 - magento/magento2#7241: No option to start with blank option for prefix and suffix in checkout. (reported by @spyrule) has been fixed in magento/magento2#11462 by @avstudnitz in 2.2-develop branch
   Related commits:
     1. 11e38f68a9956cbdc1d7c48284bc712729f43a90
     2. 7800e5dd8aac704da3f2b1799baadf2a01269c61
     3. 252529f0819d2792982452ed9d929d5fb526996d
     4. 519b805c2853f895d804c49e08bd7a23b108c38c
     5. eb79b38beb6b79411fe95d30e324d7d094ab5874
     6. fce3afe6390be6c0eb2d5fcdb2f76108d6a44e77
     7. 3e38118c1e7f92e77963a6ae4f42e66520e6f58f

srenon committed:
Add more parameters to ajax:addToCart 

The SKU by itself is not very useful because for most third-party integration you will need at minimum SKU and qty. 

The 'res' is also important because you could write a plugin for Magento\Checkout\Controller\Cart\Add to add more information to the return JSON.

ishakhsuvarov committed:
MAGETWO-85992: Throw ValidationException for invalid xml #12859

 - Merge Pull Request magento/magento2#12859 from pmclain/magento2:config-error
 - Merged commits:
   1. ab5e153aa79373da21fa9126fc4a59e5e619b5f0

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into 8830

Commit:
merge magento/2.2-develop into magento-performance/MAGETWO-82888

YevSent committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-81395

ishakhsuvarov committed:
MAGETWO-86005: [Backport 2.2] Update functional.suite.dist.yml to handle a custom backend name #12884

 - Merge Pull Request magento/magento2#12884 from scribam/magento2:2.2-mftf-backend-name
 - Merged commits:
   1. 0185572b643b273ffe74f583b55d0690945fc8e3

omiroshnichenko committed:
magento/magento2#8830: Can`t delete row in dynamicRows component
- fix case when dnd disabled

omiroshnichenko committed:
MAGETWO-84665: 8830: Can`t delete row in dynamicRows component #921

 - Merge Pull Request magento-engcom/magento2ce#921 from RomaKis/magento2:8830
 - Merged commits:
   1. 0a6a2d9b80859261afc01c3027f06babdec878aa
   2. 53c1fa324eb73394ff5edf02e3cd8c3ee8f23dab
   3. 1ab55ec53228ac6183124eb54031e7a6bdf2ec99
   4. cb241157d2405d3b7ca0cd3e36c8399d26d2888d
   5. bfb2e382b76df806f3a3307b1552dce28b6fa1eb
   6. 568d36e3d11ba4633ece4a0969680360bdf4707b
   7. 0a7289714f03276cccb1c0fbf860d85c08153abb

dvoskoboinikov committed:
Merge pull request #1877 from magento-mpi/MAGETWO-81395

Fixed issues:
- MAGETWO-81395: Provide possibility to customize error messages from payment methods

Commit:
MAGETWO-86005: [Backport 2.2] Update functional.suite.dist.yml to handle a custom backend name #12884

Commit:
MAGETWO-86016: #12712: [GitHub] Latest Google Chrome Browser issue with duplicate id… #1036

 - Merge Pull Request magento-engcom/magento2ce#1036 from serhii-balko/magento2:github-12712
 - Merged commits:
   1. 781ec89aaca6e6ee2853e98065416614f2edffce

Commit:
MAGETWO-86019: Issues/12374 #12736

 - Merge Pull Request magento/magento2#12736 from virtual97/magento2:issues/12374
 - Merged commits:
   1. a847d89917cca3f1680dbebc2611503a63fa5b1a
   2. ae6293108e4d5156a052ebcf4e249ebd06b44d51

Commit:
MAGETWO-86021: Add more parameters to ajax:addToCart #12875

 - Merge Pull Request magento/magento2#12875 from srenon/magento2:patch-6
 - Merged commits:
   1. 6d2f213efc6978f15b00105d0336a04e2a853cfb

Commit:
MAGETWO-86022: #6916 Fix notice during Update Bundle Product without changes #12734

 - Merge Pull Request magento/magento2#12734 from dzianis-yurevich/magento2:issue/6916
 - Merged commits:
   1. e6a7a4e82099de6e90239e7b0598c3ff43d1fb48

Commit:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12884: [Backport 2.2] Update functional.suite.dist.yml to handle a custom backend name (by @scribam)

ishakhsuvarov committed:
MAGETWO-85992: Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-85992-magento-magento2-12859

mslabko committed:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-82888

igrybkov committed:
Merge pull request #1880 from magento-performance/MAGETWO-82888

Fixed issue:
  - MAGETWO-82888: [Performance] Product Price indexer stuck during reindexing

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.2-develop

 - merged latest code from mainline branch

Commit:
MAGETWO-86022: #6916 Fix notice during Update Bundle Product without changes #12734

Commit:
MAGETWO-86021: Add more parameters to ajax:addToCart #12875

Commit:
MAGETWO-86016: #12712: [GitHub] Latest Google Chrome Browser issue with duplicate id… #1036

Commit:
MAGETWO-85992: Throw ValidationException for invalid xml #12859

Commit:
MAGETWO-84665: 8830: Can`t delete row in dynamicRows component #921

magento-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12734: #6916 Fix notice during Update Bundle Product without changes (by @dzianis-yurevich)
 - magento/magento2#12875: Add more parameters to ajax:addToCart (by @srenon)
 - magento-engcom/magento2ce#1036: #12712: [GitHub] Latest Google Chrome Browser issue with duplicate id… (by @serhii-balko)
 - magento/magento2#12859: Throw ValidationException for invalid xml (by @pmclain)
 - magento-engcom/magento2ce#921: 8830: Can`t delete row in dynamicRows component (by @RomaKis)


Fixed GitHub Issues:
 - magento/magento2#6916: Update Bundle Product without changes in bundle items  (reported by @magecodenet) has been fixed in magento/magento2#12734 by @dzianis-yurevich in 2.2-develop branch
   Related commits:
     1. e6a7a4e82099de6e90239e7b0598c3ff43d1fb48

 - magento/magento2#12712: Latest Google Chrome Browser issue with duplicate #email (reported by @lano-vargas) has been fixed in magento-engcom/magento2ce#1036 by @serhii-balko in 2.2-develop branch
   Related commits:
     1. 781ec89aaca6e6ee2853e98065416614f2edffce

 - magento/magento2#8830: Can`t delete row in dynamicRows component (reported by @MTheProgrammer) has been fixed in magento-engcom/magento2ce#921 by @RomaKis in 2.2-develop branch
   Related commits:
     1. 0a6a2d9b80859261afc01c3027f06babdec878aa
     2. 53c1fa324eb73394ff5edf02e3cd8c3ee8f23dab
     3. 1ab55ec53228ac6183124eb54031e7a6bdf2ec99
     4. cb241157d2405d3b7ca0cd3e36c8399d26d2888d
     5. bfb2e382b76df806f3a3307b1552dce28b6fa1eb
     6. 568d36e3d11ba4633ece4a0969680360bdf4707b
     7. 0a7289714f03276cccb1c0fbf860d85c08153abb

Commit:
MAGETWO-86019: Issues/12374 #12736

magento-team committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.2-develop

Accepted Public Pull Requests:
 - magento/magento2#12736: Issues/12374 (by @virtual97)


Fixed GitHub Issues:
 - magento/magento2#12374: Model hasDataChanges always true (reported by @Detzler) has been fixed in magento/magento2#12736 by @virtual97 in 2.2-develop branch
   Related commits:
     1. a847d89917cca3f1680dbebc2611503a63fa5b1a
     2. ae6293108e4d5156a052ebcf4e249ebd06b44d51

fooman commented:
@avstudnitz Thanks for addressing @okorshenko's feedback. Would you mind taking a look at the mentioned merge conflict as well.

avstudnitz committed:
Merge remote-tracking branch 'avs/config-set-option-share' into config-set-option-share

# Conflicts:
#	app/code/Magento/Deploy/Model/Mode.php

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=12178) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=12178) before we can accept your contribution.<br/>**10** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: Tomasz-Silpion<br/>:white_check_mark: schmengler<br/>:white_check_mark: gwharton<br/>:white_check_mark: ishakhsuvarov<br/>:white_check_mark: p-bystritsky<br/>:white_check_mark: serhii-balko<br/>:white_check_mark: okorshenko<br/>:white_check_mark: sidolov<br/>:white_check_mark: RomaKis<br/>:white_check_mark: nmalevanec<br/>:x: AzVo<br/>:x: andimov<br/>:x: duhon

avstudnitz committed:
12178 Fix unit test

avstudnitz committed:
12178 Fix code style violations

avstudnitz committed:
12178 Add missing newline at EOF

okorshenko commented:
Hi @avstudnitz 
Something goes wrong with your branch. Could you please extract your changes and apply them to new branches based on magento/magento2 : 2.2-develop
Please create new PR once ready.
Unfortunately, we can con merge this PR as is

avstudnitz commented:
Continued in https://github.com/magento/magento2/pull/13280.
